src/solvers/smt2_incremental/smt_commands.h

```diff
@@ -0,0 +1,92 @@
+// Author: Diffblue Ltd.
+
+#ifndef CPROVER_SOLVERS_SMT2_INCREMENTAL_SMT_COMMANDS_H
+#define CPROVER_SOLVERS_SMT2_INCREMENTAL_SMT_COMMANDS_H
+
+#include <solvers/smt2_incremental/smt_terms.h>
+#include <util/irep.h>
+
+class smt_commandt : protected irept
```

Comment:
As a sanity test for this, would it be worth integrating these into `smt2_solver` / `smt2_parser`? They should make the separation between parsing and solving much clearer.

Reply:
Yes. Using the new structures for the solving will help sanity test them. I have already found several minor issues by working on solving/converting to string locally. However I feel it is better to get a first version merged and then refine the structures as we work on the solving, rather than perfect the first version before merging the first PR. Integrating the new structures into the parsing would be a nice-to-have. But we are currently thinking of working on exprt -> new structures and new structures -> string before parsing.